cooltools/lib/common.py

```diff
@@ -67,3 +72,140 @@ def assign_regions_to_bins(bin_ids, regions_span):
     region_ids[mask_assigned] = regions_span["name"][bin_reg_idx_lo[mask_assigned]]
 
     return region_ids
+
+
+def download_data(name="all", cache=True, data_dir=None):
+    """
+
+    Parameters
+    ----------
+    name : str
+        Name of the dataset.
+    cache : boolean, optional
+        If True, try to load from the local cache first, and save to the cache
+        if a download is required.
+    data_dir : string, optional
+        The directory where to cache data; default is defined by :func:`get_data_dir`.
+
+    Returns
+    -------
+    Path to the last downloaded file.
+
+    """
+
+    def download_file(url, local_filename=None):
+        if local_filename is None:
+            local_filename = url.split('/')[-1]
+        print('downloading:', url, 'as', local_filename)
+        with requests.get(url, stream=True) as r:
+            r.raise_for_status()
+            with open(local_filename, 'wb') as f:
+                for chunk in r.iter_content(chunk_size=8192):
+                    f.write(chunk)
+        return local_filename
+
+    available_datasets = _get_datasets_info()
+    available_keys = map(lambda x: x[0], available_datasets)
+
+    if name in available_keys:
+        keys = [name]
+    elif name=="all":
+        keys = available_keys
+    else:
+        raise Exception(
+            """Dataset {key} is not available. 
+            Available datasets: {datasets}
+            Use print_available_datasets() to see the details. 
+            """.format(key=name, datasets=','.join(available_keys)))
+
+    data_dir = get_data_dir(data_dir)
+
+    assert len(keys)>0 # Checking that test data file is parsed successfully
+    downloaded = '' # Empty string that will be returned if the request was empty
+    for key, url, local_filename in available_datasets:
+        if not key in keys:
+            continue
+
+        file_path = os.path.join(data_dir, local_filename)
+        if cache and os.path.exists(file_path):
+            downloaded = file_path
+            continue
+        elif cache:
+            print("Test dataset {} (file {}) is not in the cache directory {}".format(key, local_filename, data_dir))
+        downloaded = download_file(url, file_path)
+
+    return downloaded
+
+
+def get_data_dir(data_dir=None):
+    """
+    Returns a path to cache directory for example datasets.
+
+    This directory is then used by :func:`download_test_data`.
```

Comment:
download_test_data was renamed to download_data

Reply:
In documentation as well, thanks!